qiskit/pulse/model/mixed_frames.py

```diff
@@ -0,0 +1,78 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Mixed Frames
+"""
+
+from .frames import Frame
+from .logical_elements import LogicalElement
+
+
+class MixedFrame:
+    """Representation of a :class:`LogicalElement` and :class:`Frame` combination.
+
+    Most instructions need to be associated with both a :class:`LogicalElement` and a :class:`Frame`.
+    The combination
+    of the two is called a mixed frame, and is represented by a :class:`MixedFrame` object.
+
+    In most cases the :class:`MixedFrame` is used more by the compiler, and a pulse program
+    can be written without :class:`MixedFrame`s, by setting :class:`LogicalElement` and
+    :class:`Frame` independently. However, in some cases using :class:`MixedFrame`s can
+    better convey the meaning of the code, and change the compilation process. One example
+    are shift/set frequency/phase instructions which are not broadcasted to other
+    :class:`MixedFrame`s if applied on a specific :class:`MixedFrame` (unlike the behavior
+    of :class:`Frame`). User can also use a subclass of :class:`MixedFrame` for a particular
```

Comment:
Why do set/shift instructions work differently for `MixedFrame` compared to `Frame`? Is there a specific use case motivating that behavior, over applying the instruction to all `MixedFrame`s sharing the `Frame`?

Reply:
The logic here is to leave all options open for the user. If you want to target a specific `MixedFrame` - you can (the same as you would target a specific `Channel` before). But, if you want to target all `MixedFrame`s associated with a `Frame` you can do that too by targeting the `Frame`.

Reply:
Oh, right, the old `Channel`s are effectively instances of `MixedFrame`s (and might become subclasses concretely?), so this preserves backwards compatibility.

What I was thinking about when asking about this is that I was thinking it made the implementation more complex since both frames and mixed frames need to be tracked in PulseIR (or other any representation). I think the way it has to work is that a frame is an alias for all of its mixed frames. So one way to think about it is that all shift/set frequency/phase instructions on a frame are replaced with those instructions on all mixed frames in a first pulse compiler pass. There is not really a standalone frame that has a frequency and phase over time, though you can think of there being one if you never apply instructions directly to mixed frames that make them different from the underlying frame.

Reply:
I think the endgame is to deprecate the `Channel`s.

Yes, one of the first passes done by the compiler would be to broadcast instructions across `MixedFrame`s. In principle, you can also imagine broadcasting instructions across `MixedFrame`s associated with a specific `LogicalElement` - I think a delay is the only reasonable example?